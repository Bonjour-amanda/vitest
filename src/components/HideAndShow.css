.HideAndShow {
    width: 80vw;
    border: 1px solid gray;
    border-radius: 5px;
}

.HideAndShow-title {
    padding: 0px 25px;
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: space-between;
    border-bottom: 1px solid gray;
}