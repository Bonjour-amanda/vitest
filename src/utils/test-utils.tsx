import { cleanup, render } from '@testing-library/react';
import React from 'react';
import { afterEach } from 'vitest';

afterEach(() => {
    cleanup();
});

const customRender = (ui: React.ReactElement, options = {}) => {
    render(ui, {
        // wrap provider(s) here if needed
    wrapper: ({ children }) => children,
    ...options,
    });
};

export * from '@testing-library/react';
export { default as userEvent } from '@testing-library/user-event';

export { customRender as render };