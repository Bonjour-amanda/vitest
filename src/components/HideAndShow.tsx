import React, { useState } from "react";
import './HideAndShow.css'

type HideAndShowProps = {
    title: string;
    children: React.ReactNode;
}
const HideAndShow = (props: HideAndShowProps) => {

    const {title, children} = props;
    const [show, setShow] = useState(false);

    const onAccordionClick = () => {
        setShow(!show);
    }

    return (
        <div className="HideAndShow">
            <div className="HideAndShow-title">
                <h3>{title}</h3>
                <button onClick={() => onAccordionClick()}>{!show ? 'Visible' : 'Hide'}</button>
            </div>
            {show && (
                <div>
                    {children}
                </div>
            )}
        </div>
    );
}

export default HideAndShow;