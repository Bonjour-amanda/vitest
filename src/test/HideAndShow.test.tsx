import { beforeEach, describe, expect, test, it } from "vitest";
import { fireEvent, waitFor } from '@testing-library/react';
import HideAndShow from "../components/HideAndShow";
import '@testing-library/jest-dom';
import { act } from 'react-dom/test-utils';
import { render, screen, userEvent } from '../utils/test-utils';

describe('Hide and Show Component', () => {

    beforeEach(() => {
        render(<HideAndShow title='Hide and Show Testing'><h4>Content</h4></HideAndShow>)
    })
    
    test('Hide and show component snapshot test', () => {
        const component = render(<HideAndShow title='Hide and Show Testing'><h4>Content</h4></HideAndShow>);
      
        expect(component).toMatchSnapshot()
    })

    it('should show title', () => {
        expect(screen.getByText(/Hide and Show Testing/i)).toBeInTheDocument();
    })


    test("should not show the content at the start", () => {

        expect(screen.queryByText(/Content/i)).not.toBeInTheDocument();
    })

    test("should show the content on accordion click",async () => {

        const title = screen.getByText(/Visible/i);
        fireEvent.click(title)

        expect(await screen.findByText(/Content/i)).toBeInTheDocument();
    })
})
