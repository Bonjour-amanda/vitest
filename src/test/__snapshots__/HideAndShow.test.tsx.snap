// Vitest Snapshot v1

exports[`Hide and Show Component > Hide and show component snapshot test 1`] = `undefined`;
